op-batcher/batcher/channel_manager.go

```diff
@@ -344,3 +355,27 @@ func l2BlockRefFromBlockAndL1Info(block *types.Block, l1info derive.L1BlockInfo)
 		SequenceNumber: l1info.SequenceNumber,
 	}
 }
+
+// Close closes the current pending channel, if one exists, and prevents the
+// creation of any new channels.
+// This ensures that no new blocks will be added to the channel, but there may be any number
+// of frames still produced by calling `OutputFrames()`, which flushes the compression buffer.
+// These frames still need to be published.
+func (s *channelManager) Close() {
+	if s.closed {
+		return
+	}
+
+	s.closed = true
+
+	// Any pending state can be proactively cleared if there are no submitted transactions
+	if len(s.confirmedTransactions) == 0 && len(s.pendingTransactions) == 0 {
+		s.clearPendingChannel()
+	}
+
+	if s.pendingChannel == nil {
+		return
+	}
+
+	s.pendingChannel.Close()
```

Comment:
see previous comment, we might consider adding 
```
return s.outputFrames()
```
here (would then need to add error return to `Close`) and then early-exit in `TxData` when closed.